services-geocoding/src/main/java/com/mapbox/api/geocoding/v5/models/RoutablePoint.java

```diff
@@ -0,0 +1,106 @@
+package com.mapbox.api.geocoding.v5.models;
+
+import androidx.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.geojson.Point;
+
+/**
+ * Object representing {@link CarmenFeature}'s routable point.
+ *
+ * @since 6.10.0
+ */
+@AutoValue
+public abstract class RoutablePoint {
+
+  /**
+   * A string representing the routable point name.
+   *
+   * @return routable point name
+   * @since 6.10.0
+   */
+  @Nullable
```

Comment:
Is this really nullable? I mean, what I'd expect is `RoutablePoints` to be nullable but If they're not, any `RoutablePoint` from the list to have valid values. Does that make sense?

Reply:
@Guardiola31337 Do you mean by duplicating e.g. the rooftop location in `RoutablePoints`? If yes, I disagree.

I assume the correct distinction would be to leave `RoutablePoints` null if there are none and let the app logic decide how to behave in that case (e.g. fallback to original location). Otherwise it's becoming ambiguous whether the data is actually available for the specific search result.

Reply:
@Guardiola31337 in addition to what @kalbaxa said, I think it's better to leave it as nullable because all the `service-geocoder` relies on `AutoValue/GSON` serialisation/deserialisation. If we declare any field as non-nullable and for some reason backend don't send the value, the SDK will crash. I don't want this. I think it's better to check for null on customer's side and use alternative fallback. Also, nullable fields are consistent with the current approach.

Reply:
@kalbaxa What I meant was that routable points could be `null` (that's totally fine) but if the server emits a routable point, I would expect not to be `null` (having `null` fields), that's the contract I'd expect from the backend.

Reply:
@DzmitryFomchyn IMO errors in the backend should fail-fast, this approach would hide this type of issues as they'd go unnoticed.

Reply:
A can agree with that, but we should also be consistent here. All the current code relies on nullability, it won't make any better if we make one field as non-null while everything else remain nullable. Given everything that's written above, I think we should leave as is now and discuss another approach in case of SDK refactoring, if it happen. More likely, `service-geocoder` will be deprecated soon.

Reply:
> If we declare any field as non-nullable and for some reason backend don't send the value, the SDK will crash. I don't want this.

+++

Event if it's not an error on the backend side, it can be a logic change. For example, we accidentally made waypoints nullable but that played well because after some time they did become nullable because a case was introduced that they can be returned in a different response level.

Reply:
> it can be a logic change

A logic change like this should mean a major version upgrade of an endpoint. All customers (not only Nav SDK) would be impacted by such an intentional change.

Reply:
Not necessarily. With waypoints they wouldn't (and weren't). Because it depends on a new request parameter which falls back to old behaviour.

Reply:
We would have to provide backwards compatibility if that wasn't the case (either on backend or mapbox-java level). We were lucky and needed less work for that specific situation but that shouldn't be the reason to make contracts less strict going forward.